packages/core/src/model.ts

```diff
@@ -139,9 +139,54 @@ export class ChartModel {
 			dataGroups
 		});
 
+		if (this.getZoomBarData()) {
```

Comment:
shouldn't there also be a check for whether the zoombar is enabled?

Reply:
Also, it seems like we need a new model class for axis charts. We can call it `AxisChartsModel` and it can extend `Model` & add all this stuff. There's no zoombar in a pie chart for instance so it doesn't make sense to include all this for it

Reply:
unfortunately this has to live here for now I think because I need to update the zoomBarData whenever the models setData method is called (to sync it), the only way that this.getZoomBarData() returns with values if it a zoom-bar.ts has already set it up